src/facade/acl_commands_def.h

```diff
@@ -4,8 +4,24 @@
 
 #pragma once
 
+#include <cstdint>
+#include <limits>
+#include <string>
+#include <utility>
+#include <vector>
+
 namespace dfly::acl {
 // Special flag/mask for all
 constexpr uint32_t NONE = 0;
 constexpr uint32_t ALL = std::numeric_limits<uint32_t>::max();
+
+enum class KeyOp : int8_t { READ, WRITE, READ_WRITE };
+
+using GlobType = std::pair<std::string, KeyOp>;
+
+struct AclKeys {
+  std::vector<GlobType> key_globs;
```

Comment:
@dranikpg This is fine. We agreed we would implement this the naive way. There will be overhead but only when the keys are actually used. (You will see when I push the validation).

Reply:
That's clear there won't be any overhead without keys, but I'm surprised you keep it raw.

For example, if we only support prefix / suffix, aka `*word*` (which covers a lot), you can then check easily in linear time (by string size / max prefix/suffix len) with a simple trie or hashes. It doesn't take much effort

Reply:
We think it's premature optimisation. Feature complete is what we are after here